drivers/adc/adc_cc13xx_cc26xx.c

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright (c) 2022 Florin Stancu <niflostancu@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_cc13xx_cc26xx_adc
+
+#include <errno.h>
+
+#define LOG_LEVEL CONFIG_ADC_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(adc_cc13xx_cc26xx);
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <zephyr/drivers/adc.h>
+#include <soc.h>
+
+/* Driverlib includes */
+#include <inc/hw_types.h>
+#include <driverlib/interrupt.h>
+#include <driverlib/ioc.h>
+#include <driverlib/rom.h>
+#include <driverlib/prcm.h>
+#include <driverlib/aux_adc.h>
+#include <ti/devices/cc13x2_cc26x2/inc/hw_aux_evctl.h>
+
+#define ADC_CONTEXT_USES_KERNEL_TIMER
+#include "adc_context.h"
+
+
+/**
+ * Channels are based on ADC_COMPB_IN_* hal_ti definitions, max. index is 16 (included).
+ */
+#define CHAN_COUNT 0x10
```

Comment:
This is a bit misleading, as in fact the channel count is 17. Perhaps something like `MAX_CHAN_ID` would be more adequate.